Readability-readerable.js

```diff
@@ -55,6 +55,21 @@ function isProbablyReaderable(doc, options = {}) {
 
   var nodes = doc.querySelectorAll("p, pre");
 
+  // Get <div> nodes which meets certain criteria them into the `nodes` variable.
+  var set = new Set(nodes);
+  var divs = doc.querySelectorAll("div");
+  if (divs.length) {
+    [].forEach.call(divs, function (node) {
+      // add the parent node of divs that have no child elements and contain a small amount of text
+      if (!node.children.length && node.textContent.length < options.minContentLength) {
+        set.add(node.parentNode);
+      // divs that only contain inline text
+      } else if (![].some.call(node.children, child => !["A", "BR", "SPAN", "PRE"].includes(child.nodeName))) {
```

Comment:
This would include divs that contain only links, and would also include (all) divs that have no children at all, or no text at all (though they're excluded from the previous `if` condition). I don't think that's a good idea, because it's likely to include items that are tables of content or "see also" references or such. Article text shouldn't (ideally) be full of links. We could extend this check to ensure that there is some other, non-link, non-empty text nodes, perhaps? Instead of only looping over the children, we could add a helper function that loops over `childNodes`, returns `false` as soon as it finds a non-inline child, and returns `true` as soon as it's seen enough inline text to qualify the node? For `a`, `span` and `font` tags we could determine the `textContent` length of the inner child. I'm not sure why `pre` is in the list here...

A nitpick on the actual code: Given how hot this code is, putting the set of inline nodes into a set and using `inlineSet.has(child.nodeName)` seems like a worthwhile optimization.